src/handlers/opportunitiesHandlers.ts

```diff
@@ -1,89 +1,52 @@
-import { ajv } from '../ajv';
-import { getLogger } from '../logger';
-import { db } from '../database';
-import { isTinyPgErrorWithQueryContext } from '../types';
 import {
-	DatabaseError,
-	InputValidationError,
-	InternalValidationError,
-} from '../errors';
+	createOpportunity,
+	getLimitValues,
+	loadOpportunity,
+	loadOpportunityBundle,
+} from '../database';
+import {
+	isId,
+	isTinyPgErrorWithQueryContext,
+	isWritableOpportunity,
+} from '../types';
+import { DatabaseError, InputValidationError } from '../errors';
+import { extractPaginationParameters } from '../queryParameters';
 import type { Request, Response, NextFunction } from 'express';
-import type { Result } from 'tinypg';
-import type { JSONSchemaType } from 'ajv';
-import type { Opportunity } from '../types';
-
-const logger = getLogger(__filename);
 
 const getOpportunities = (
 	req: Request,
 	res: Response,
 	next: NextFunction,
 ): void => {
-	db.sql('opportunities.selectAll')
-		.then((opportunitiesQueryResult: Result<Opportunity>) => {
-			logger.debug(opportunitiesQueryResult);
-			const { rows: opportunities } = opportunitiesQueryResult;
-			res.status(200).contentType('application/json').send(opportunities);
+	const paginationParameters = extractPaginationParameters(req);
+	loadOpportunityBundle({
+		...getLimitValues(paginationParameters),
+	})
+		.then((opportunityBundle) => {
+			res.status(200).contentType('application/json').send(opportunityBundle);
 		})
 		.catch((error: unknown) => {
 			if (isTinyPgErrorWithQueryContext(error)) {
-				next(new DatabaseError('Error retrieving opportunity.', error));
+				next(new DatabaseError('Error retrieving opportunities.', error));
 				return;
 			}
 			next(error);
 		});
 };
 
-interface GetOpportunityParams {
-	id: number;
-}
-const getOpportunityParamsSchema: JSONSchemaType<GetOpportunityParams> = {
-	type: 'object',
-	properties: {
-		id: {
-			type: 'integer',
-			minimum: 1,
-		},
-	},
-	required: ['id'],
-};
-const isGetOpportunityParams = ajv.compile(getOpportunityParamsSchema);
 const getOpportunity = (
-	req: Request<unknown, unknown, GetOpportunityParams>,
+	req: Request,
 	res: Response,
 	next: NextFunction,
 ): void => {
-	if (!isGetOpportunityParams(req.params)) {
-		next(
-			new InputValidationError(
-				'Invalid request parameters.',
-				isGetOpportunityParams.errors ?? [],
-			),
-		);
+	const { opportunityId } = req.params;
+	if (!isId(opportunityId)) {
```

Comment:
I tried `GET opportunities/555555555555555555555555555555` and got this response:
```json
{
  "name": "DatabaseError",
  "message": "Unexpected database error.",
  "details": [
    {
      "length": 168,
      "name": "error",
      "severity": "ERROR",
      "code": "22003",
      "where": "unnamed portal parameter $1 = '...'",
      "file": "numutils.c",
      "line": "314",
      "routine": "pg_strtoint32"
    }
  ]
}
```

Which is fine and probably not a new error at all, but would be nice to make it a 400 since we don't support IDs greater than 2^32-1.

Reply:
Opening an issue!  The good news is this will be easy to change across *all* endpoints due to the `getId` type guard / validator being reused.

Reply:
Done: https://github.com/PhilanthropyDataCommons/service/issues/971